src/mobile/src/main/java/tupperdate/android/home/Home.kt

```diff
@@ -1,98 +1,146 @@
 import androidx.ui.tooling.preview.Preview
+import tupperdate.android.appbars.mainBottomBar
+import tupperdate.android.appbars.mainTopBar
 import tupperdate.android.ui.TupperdateTheme
-import tupperdate.api.AuthenticationApi
+import tupperdate.api.RecipeApi
+import tupperdate.api.api
+import kotlin.math.absoluteValue
 
 @Composable
+/*
+@ Guy-Laurent : I am forced to do only one function because it is the only way
+to call like() and dislike() without the weird RealRecipeApi import.
+ */
 fun Home(
-    onButtonClick: () -> Unit,
-    user: AuthenticationApi.Profile?,
-    modifier: Modifier = Modifier,
+    recipeApi: RecipeApi,
+    onChatClick: () -> Unit,
+    onProfileClick: () -> Unit,
+    onReturnClick: () -> Unit,
+    onRecipeClick: () -> Unit
 ) {
-    if (user == null) {
-        HomeDisconnected(
-            onButtonClick,
-            modifier,
-        )
-    } else {
-        HomeConnected(
-            onButtonClick,
-            user,
-            modifier,
-        )
-    }
-}
+    // TODO make a call to front element of recipeApi
+    val presentRecipe = (RecipeApi.Recipe(
+        "Red lobster",
+        "Look at me, am I not yummy ?",
+        "lobster.jpg"
+    ))
 
-@Composable
-private fun HomeDisconnected(
-    onButtonClick: () -> Unit,
-    modifier: Modifier,
-) {
-    Column(
-        modifier.padding(10.dp)
-    ) {
-        Text(text = "Hi there.")
-        Button(onClick = onButtonClick) {
-            Text("Go to onboarding")
-        }
-    }
+    mainTopBar(onChatClick = onChatClick, onProfileClick = onProfileClick)
+
+    DisplayRecipeCard(
+        presentRecipe = presentRecipe,
+        onLike = { recipeApi.like(presentRecipe) },
+        onDislike = { recipeApi.dislike(presentRecipe) },
+        onReturnClick = onReturnClick,
+        onRecipeClick = onRecipeClick
+    )
 }
 
 @Composable
-private fun HomeConnected(
-    onButtonClick: () -> Unit,
-    user: AuthenticationApi.Profile,
-    modifier: Modifier,
+private fun DisplayRecipeCard(
+    presentRecipe: RecipeApi.Recipe,
+    onLike: () -> Unit,
+    onDislike: () -> Unit,
+    onReturnClick: () -> Unit,
+    onRecipeClick: () -> Unit
 ) {
-    Column(
-        modifier.padding(10.dp)
-    ) {
-        Text(text = "Hi there ${user.displayName}")
-        Button(onClick = onButtonClick) {
-            Text("View Branding")
+    Row(Modifier.fillMaxSize()) {
+        Column(
+            Modifier.fillMaxWidth().align(Alignment.Bottom)
+                .padding(bottom = veryLittleButtonSize.dp)
+        ) {
+            WithConstraints { -> //why no parameter here ??
+
+                //determine height and width of card according to constraints
+                //the card measures 90 % of screen width and 75 % of screen height
+                val boxHeight =
+                    with(DensityAmbient.current) { constraints.maxHeight.toDp() * 0.75f }
+                val boxWidth = with(DensityAmbient.current) { constraints.maxWidth.toDp() * 0.9f }
```

Comment:
The card should fit its content size.

Reply:
maxWidth is the size of the screen. We don't want the card to occupy it entierly.